SoftLayer/CLI/call_api.py

```diff
@@ -2,20 +2,31 @@
 
 import click
 
+import json
+
 from SoftLayer.CLI import environment
 from SoftLayer.CLI import formatting
 
 
+def validate_filter(ctx, param, value):
+    try:
+        if value:
+            return json.loads(value)
+    except ValueError:
+        raise click.BadParameter('filters need to be in JSON format')
+
+
 @click.command('call', short_help="Call arbitrary API endpoints.")
 @click.argument('service')
 @click.argument('method')
 @click.argument('parameters', nargs=-1)
 @click.option('--id', '_id', help="Init parameter")
+@click.option('--filter', '_filter', callback=validate_filter, help="Object filter in a JSON string")
```

Comment:
This line is a bit too long. Can you break the arguments into their own lines? Like this:

```python
@click.option('--filter', '_filter',
			  callback=validate_filter,
			  help="Object filter in a JSON string")
```